arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json

```diff
@@ -0,0 +1,1744 @@
+          "description": "The SKU of the container registry."
+        },
+        "properties": {
+          "$ref": "#/definitions/RegistryProperties",
+          "description": "The properties of the container registry.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "Sku": {
+      "description": "The SKU of a container registry.",
+      "required": [
+        "name"
+      ],
+      "properties": {
+        "name": {
+          "description": "The SKU name of the the container registry. Required for registry creation.",
+          "enum": [
+            "Basic",
+            "Standard",
+            "Premium"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "SkuName",
+            "modelAsString": true
+          }
+        },
+        "tier": {
+          "description": "The SKU tier based on the SKU name.",
+          "enum": [
+            "Basic",
+            "Standard",
+            "Premium"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "SkuTier",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "RegistryProperties": {
+      "description": "The properties of a container registry.",
+      "properties": {
+        "loginServer": {
+          "description": "The URL that can be used to log into the container registry.",
+          "type": "string",
+          "readOnly": true
+        },
+        "creationDate": {
+          "format": "date-time",
+          "description": "The creation date of the container registry in ISO8601 format.",
+          "type": "string",
+          "readOnly": true
+        },
+        "provisioningState": {
+          "description": "The status of the container registry at the time the operation was called.",
+          "enum": [
+            "Creating",
+            "Succeeded"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": false
+          }
+        },
+        "adminUserEnabled": {
+          "description": "The value that indicates whether the admin user is enabled.",
+          "default": false,
+          "type": "boolean"
+        },
+        "storageAccount": {
+          "$ref": "#/definitions/StorageAccountProperties",
+          "description": "The properties of the storage account for the container registry. Only applicable for Basic SKU."
+        }
+      }
+    },
+    "StorageAccountProperties": {
+      "description": "The properties of a storage account for a container registry. Only applicable for Basic SKU.",
+      "properties": {
+        "name": {
+          "description": "The name of the storage account.",
+          "type": "string"
+        }
+      }
+    },
+    "RegistryCreateParameters": {
+      "description": "The parameters for creating a container registry.",
```

Comment:
You could still do an allOf on ```Resource``` here, since the ```name```, ```id``` and ```type``` properties are readonly they won't be a part of the request.